include/common/array.h

```diff
@@ -41,4 +41,17 @@ wl_array_len(struct wl_array *array)
 		pos && (const char *)pos >= (const char *)(array)->data;               \
 		(pos)--)
 
+/**
+ * xwl_array_add() - add item to array and crash on allocation error
+ * @_arr: wl_array to add the item to
+ * @_val: the item to add to the array
+ * Note: asserts() when the memory allocation for the new item failed
+ */
+#define xwl_array_add(_arr, _val) do {                     \
```

Comment:
This is indeed a nice wrapper. I now get it :smile:

Suggest that we avoid `assert()` and just `exit()` with an error message.

Suggest that we call it `array_add()` becase:
1. `xwl` makes me think of XWayland
2. It actually does more than wl_array_add() + die-on-null

Also suggest adding the following to the comment in the header to make it easier to wrap one's head around it:

```
Let us illustrate the function of this macro by an example:

    uint32_t value = 5;
    array_add(array, value);

...is the equivalent of the code below which is how you would
otherwise use the wl_array API:

uint32_t *elm = wl_array_add(array, sizeof(uint32_t));
    if (!elm) {
    	perror("failed to allocate memory");
    	exit(EXIT_FAILURE);
    }
    *elm = value;
```

Reply:
Renamed to `array_add()`, replaced the assert with `exit()` and added your comment.